CanonicalLayouts/supporting-pane-compose/app/src/main/java/com/example/supportingpanecompose/ui/SupportingPaneSample.kt

```diff
@@ -47,53 +54,82 @@ private val data = mapOf(
-        main = {
-            Column(
-                horizontalAlignment = Alignment.CenterHorizontally,
-                modifier = Modifier.fillMaxSize()
+    SupportingPaneScaffold(
+        supportingPane = {
+            AnimatedPane(
+                modifier = Modifier
+                    .background(
+                        color = Color.LightGray.copy(alpha = .3f),
+                        shape = RoundedCornerShape(16.dp)
+                    )
+                    .padding(all = 16.dp)
             ) {
-                Text("Main Content", style = MaterialTheme.typography.titleLarge)
-                Text(selectedTopic)
-            }
-        },
-        supporting = {
-            Column(
-                modifier = Modifier.fillMaxSize()
-            ) {
-                Text("Related Content", style = MaterialTheme.typography.titleLarge)
+                Column {
+                    Text(
+                        "Related Content",
+                        modifier = Modifier.padding(vertical = 16.dp),
+                        style = MaterialTheme.typography.titleLarge
+                    )
 
-                LazyColumn {
-                    items(
-                        data.getValue(selectedTopic),
-                        key = { it }
-                    ) { relatedTopic ->
-                        Box(
-                            Modifier
-                                .fillMaxWidth()
-                                .clickable {
-                                    selectedTopic = relatedTopic
-                                }
-                        ) {
-                            Text(
-                                text = relatedTopic,
-                                modifier = Modifier
-                                    .padding(16.dp)
-
-                            )
+                    LazyColumn {
+                        items(
+                            data.getValue(selectedTopic),
+                            key = { it }
+                        ) { relatedTopic ->
+                            Box(
+                                Modifier
+                                    .fillMaxWidth()
+                                    .padding(all = 4.dp)
+                                    .clickable {
+                                        selectedTopic = relatedTopic
+                                        if (navigator.canNavigateBack()) {
+                                            navigator.navigateBack()
+                                        }
+                                    }
+                            ) {
+                                Text(
+                                    text = relatedTopic,
+                                    modifier = Modifier
+                                )
+                            }
                         }
                     }
                 }
+
+                BackHandler(enabled = navigator.canNavigateBack()) {
```

Comment:
Nit - even if here works fine, I think it makes more sense to put this `BackHandler` right below `val navigator`.

There doesn't feel like a reason why it should go in the `supportingPane` slot or the main content slot, so bringing it up to where the navigator is declared feels more prominent to me.